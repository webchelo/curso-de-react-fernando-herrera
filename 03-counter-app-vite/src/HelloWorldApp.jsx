export const HelloWorldApp = () => {
    // document.createElement...
    return (
        <h1>Hola Mundo!</h1>
    );
}