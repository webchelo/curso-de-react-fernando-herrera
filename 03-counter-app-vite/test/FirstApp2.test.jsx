import { render, screen } from "@testing-library/react"
import { FirstApp } from "../src/FirstApp"

describe('Pruebas en <FirstApp />', () => {
    const title = 'Hola, soy chelo'
    const subTitle=  'Soy un subtitulo'
    test('debe hacer match con el snapshot', () => {
        const { container } = render( <FirstApp title={title}/>)
        expect(container).toMatchSnapshot();
    })
    test('debe mostar el mensaje "Hola, soy chelo"', () => {
        render( <FirstApp title={title}/>)
        
        expect(screen.getByText(title)).toBeTruthy();
        

    })
    test('debe mostrar el titulo en un h1', () => {
        render( <FirstApp title={title}/>)
        expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title);
    })
    test('debe de mostrar el subtitulo enviado por props', () => {
        render( <FirstApp
            title={title}
            subTitle={subTitle}
        />)
        expect(screen.getAllByText(subTitle).length).toBe(2)
    })
 })