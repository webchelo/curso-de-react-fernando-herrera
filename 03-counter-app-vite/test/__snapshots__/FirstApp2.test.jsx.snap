// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy chelo
  </h1>
  <p>
    0
  </p>
  <p>
    0
  </p>
  <p>
    No hay nombre
     
  </p>
</div>
`;
